libpff/libpff_local_descriptors.c

```diff
@@ -371,6 +371,7 @@ int libpff_local_descriptors_get_leaf_node_from_node_by_identifier(
 	{
 		return( 0 );
 	}
+    result = 0;
```

Comment:
This variable is used twice - just above, it's the return value of a submethod, but then it's also used later to represent the return value of this method itself. In between, we should make sure to reset it so that the first value doesn't accidentally get used for the second... this method is searching for a matching node and a value of `0` means no such node was found, so it's an appropriate default.